src/components/map/rent-popup.tsx

```diff
@@ -150,6 +160,19 @@ const RentPopup: React.FC<RentPopupProps> = ({
     [openedStationId],
   );
 
+  useEffect(
+    () => {
+      if (booking && openedStationId && stationDetail && openedStationId === booking.stationId) {
```

Comment:
Please use early return to reduce the nesting here.

Reply:
See 9a6d72c29f9b069a5c834c91b7dca3e174c473d5.